### ISSUE TITLE
reportUnusedVariable false positives

### ISSUE THREAD

**Issue by bgrundmann** [microsoft/pyright#3728]
**Describe the bug**
reportUnusedVariables reports used variables (found in variables captured by a match statement).  This is a regression in 1.1.262

**To Reproduce**

```python
# pyright: strict

def test():
    match [1,2]:
        case [a, b]:
            print(a,b) 

        case _:
            print("other")
```

**Output of 1.1.261 (aka expected)**

```
» pyright bug_pyright.py 
WARNING: there is a new pyright version available (v1.1.261 -> v1.1.262).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

No configuration file found.
pyproject.toml file found at /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core.
Loading pyproject.toml file at /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/pyproject.toml
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/typings is not a valid directory.
Searching for source files
Found 1 source file
pyright 1.1.261
0 errors, 0 warnings, 0 informations 
Completed in 0.617sec
```

**Output of 1.1.262 (aka the bug)**

```
» pyright bug_pyright.py 
No configuration file found.
pyproject.toml file found at /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core.
Loading pyproject.toml file at /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/pyproject.toml
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/typings is not a valid directory.
Searching for source files
Found 1 source file
pyright 1.1.262
/Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/bug_pyright.py
  /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/bug_pyright.py:5:15 - error: Variable "a" is not accessed (reportUnusedVariable)
  /Users/benediktgrundmann/SynologyDrive/Programming/localzero/localzero-generator-core/bug_pyright.py:5:18 - error: Variable "b" is not accessed (reportUnusedVariable)
2 errors, 0 warnings, 0 informations 
Completed in 0.617sec
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.263, which I just published.

**Comment by bgrundmann** [comment]
I can confirm that the fix works. Thanks.
